### DISCUSSION TITLE
charon - receive_delay_type configuration parameter

### DISCUSSION THREAD

**Discussion by petitsyeux** [strongswan/strongswan#1463] [Q&A]
Hi,
What is the expected format of the _receive_delay_type_ parameter in the charon.conf configuration file?
I need to delay a bit the reception of a CHILD_DELETE request. Is it possible to use this receive_delay_type parameter to do what I want?
Thanks for your feedback
s.

**Comment by tobiasbrunner** [comment]
As documented, it takes the numeric IKE message/exchange type of the messages to delay. Since there is no CHILD_DELETE request as such, but rather a DELETE payload in an INFORMATIONAL request, you can't directly target the deletion of a CHILD_SA with this. But depending on your goal and the test scenario, delaying INFORMATIONAL requests might work (`receive_delay_request = no` and `receive_delay_type = 37`). What exactly are you testing?

**Reply by petitsyeux** [reply]
Hi Tobias,
We have about 40 embedded systems using strongswan 5.4.0 (we will upgrade soon) that connect to a Palo Alto firewall. Since an update of the firewall, we are experiencing random _CHILD_REKEY collision with CHILD_DELETE_ that drop our phase2 during some rekeying. 
The initial problem is that the Palo Alto sometimes returns a DELETE request before initializing the new CHILD_SA which leads to a collision. As we don't have hand on the firewall configuration, I look for a workaround on the charon configuration side and I would like to try to delay CHILD_REKEY requests from the Palo Alto to avoid collisions. 

```
Oct 14 09:01:41 svom daemon.info swan: 15[CFG] received stroke: rekey '{19}'
Oct 14 09:01:41 svom daemon.info swan: 15[IKE] queueing CHILD_REKEY task
Oct 14 09:01:41 svom daemon.info swan: 15[IKE] activating new tasks
Oct 14 09:01:41 svom daemon.info swan: 15[IKE]   activating CHILD_REKEY task
Oct 14 09:01:41 svom daemon.info swan: 15[IKE] establishing CHILD_SA conn{2}
Oct 14 09:01:42 svom daemon.info swan: 08[IKE] received DELETE for ESP CHILD_SA with SPI b9dcbeca
Oct 14 09:01:42 svom daemon.info swan: 08[IKE] closing CHILD_SA conn{19} with SPIs c7aa6641_i (0 bytes) b9dcbeca_o (0 bytes) and TS ip1/32 === ip2/27
Oct 14 09:01:42 svom daemon.info swan: 08[IKE] sending DELETE for ESP CHILD_SA with SPI c7aa6641
Oct 14 09:01:42 svom daemon.info swan: 08[IKE] CHILD_SA closed
Oct 14 09:01:42 svom daemon.info swan: 08[IKE] detected CHILD_REKEY collision with CHILD_DELETE
Oct 14 09:01:42 svom daemon.info swan: 13[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED notify
Oct 14 09:01:42 svom daemon.info swan: 13[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
Oct 14 09:01:42 svom daemon.info swan: 13[IKE] CHILD_SA conn{20} established with SPIs c6937174_i a24c0cc3_o and TS ip1/32 === ip2/27
Oct 14 09:01:42 svom daemon.info swan: 13[IKE] CHILD_SA rekey/delete collision, deleting redundant child
Oct 14 09:01:42 svom daemon.info swan: 13[IKE] reinitiating already active tasks
Oct 14 09:01:42 svom daemon.info swan: 13[IKE]   CHILD_REKEY task
Oct 14 09:01:42 svom daemon.info swan: 13[IKE] closing CHILD_SA conn{20} with SPIs c6937174_i (0 bytes) a24c0cc3_o (0 bytes) and TS ip1/32 === ip2/27
Oct 14 09:01:42 svom daemon.info swan: 13[IKE] sending DELETE for ESP CHILD_SA with SPI c6937174
Oct 14 09:01:43 svom daemon.info swan: 14[IKE] received DELETE for ESP CHILD_SA with SPI a24c0cc3
Oct 14 09:01:43 svom daemon.info swan: 14[IKE] CHILD_SA closed
Oct 14 09:01:43 svom daemon.info swan: 14[IKE] activating new tasks
Oct 14 09:01:43 svom daemon.info swan: 14[IKE] nothing to initiate
```

**Reply by tobiasbrunner** [reply]
Is that related to #1458? Anywa, you should definitely report the issue to Palo Alto as their implementation behaves incorrectly.

> The initial problem is that the Palo Alto sometimes returns a DELETE request before initializing the new CHILD_SA which leads to a collision.

As rekey responder it should not send a DELETE at all (until the initiator deletes the old SA).

**Reply by petitsyeux** [reply]
Yes it seems to be related to the same issue. We have open tickets with Palo Alto support but the exchanges are not very responsive so I'm looking for a workaround at the moment :)
I'm sorry, but I've been looking in the strongswan documentation but I can't find the numeric IKE message/exchange type of the messages. Is this information in the strongswan doc (url please) or in the RFC?
Thank you Tobias for your help and for your work on strongswan.

**Reply by petitsyeux** [reply]
OK Tobias. I found the information inside the RFC.
